### ISSUE TITLE
Is this library still working?

### ISSUE THREAD

**Issue by vineshhpatel** [alkawryk/tinderjs#28]
I tried the auth method with facebook token and facebook user id and got error.

`node_modules/tinderjs/tinder.js:166
          throw "Failed to authenticate: " + body.error
                                           ^
Failed to authenticate: Access Denied`

I also tried with CURL.

`curl -X POST https://api.gotinder.com/auth --data '{"facebook_token": CAAG..., "facebook_id": 1627...}'`

`{"code":401,"error":"FacebookTokenRequired"}`

Is there anything that I am missing?

**Comment by jensvdh** [comment]
Wondering the same

**Comment by fregante** [comment]
You might have the wrong credentials. It works great for me.

**Comment by vineshhpatel** [comment]
@bfred-it can you share your code if you can :))

**Comment by fregante** [comment]
This is my code.

```
var tinder = require('tinderjs');
var client = new tinder.TinderClient();
client.authorize(TINDER_USER_TOKEN, TINDER_USER_ID, function () {
    console.log(arguments);
});
```

To get the `TINDER_USER_TOKEN` and `TINDER_USER_ID`, use a proxy: https://github.com/tranhungt/tinder_pro#getting-the-facebook-oauth-tokens

Once you have your user ID, you can just get the updated token by visiting this url and getting the token from the URL before it disappears: https://www.facebook.com/dialog/oauth?client_id=464891386855067&redirect_uri=https://www.facebook.com/connect/login_success.html&scope=basic_info,email,public_profile,user_about_me,user_activities,user_birthday,user_education_history,user_friends,user_interests,user_likes,user_location,user_photos,user_relationship_details&response_type=token

I know that you can also set up a new Facebook application to avoid using a proxy and that method, but I never tried it.

**Comment by vineshhpatel** [comment]
@bfred-it Thanks !!! I will try this. :+1:

**Comment by frederocks** [comment]
@bfred-it can you suggest a way to programmatically access the token from a request object?

**Comment by fregante** [comment]
@frederocks I wish I knew. Some articles just suggest creating a Facebook app and using the official APIs for this.

**Comment by sarment0** [comment]
I tried using Facebook SDK, and it gives me a not working Access token.
I think it's because of the "client_id" you must use this: 464891386855067
I guess it's the Tinder APP ID on facebook.... 

This could be functional if i could login in the application programmatically, i tried to copy the content of the opened window but the addEventListener is not exposed to the opener window... So i don't have access to read the content... 

I'm frustrated i have to no idea what to do.

**Comment by mayeaux** [comment]
@sarment0 

Ping me here : https://tinderjs.slack.com/ and I will help you get it set up :)

**Comment by mayeaux** [comment]
@sarment0 sorry just realized that's not a public slack channel!

Open an issue here:

https://github.com/tinderjs/tinderjs

And I'll get you set up :)

**Comment by mayeaux** [comment]
Just published a library that will let you get your Facebook token and id programatically :) https://github.com/tinderjs/tinderauth

**Comment by sarment0** [comment]
@mrmayfield  Great, will test. (I don't like the idea of given the username and password... but i will give the shot).

Tks

**Comment by mayeaux** [comment]
As mentioned in the docs there, for this kind of Tinder hacking stuff I would use a throwaway Facebook account as opposed to a real one anyways.

If you want to try this , I fixed the tindercred module so you should be able to use it now to get your credentials from here if you don't want to share your password:

https://github.com/tinderjs/tindercred

Let me know if you can get that working :)

**Comment by sarment0** [comment]
@mrmayfield It worked perfectly.
I will try using this in my project.

**Comment by mayeaux** [comment]
Check out and use http://www.github.com/tinderjs/tinderjs  , I've improved the library a lot and merged most of the open PRs. You can open issues there if you run into issues we have a few people in the organization who can help you.

**Comment by dwarrier** [comment]
I'm trying to use tindercred to get the credentials, and then use cURL to access the tinder API. For some reason it isn't working (I'm getting the same error as sarment0). I'm using:

```
curl -X POST https://api.gotinder.com/auth --data '{"facebook_token": CAAG..., "facebook_id": 777...}'
```

Where the values are taken directly from tindercred. Any tips on what I may be doing wrong?

**Comment by mayeaux** [comment]
@dwarrier , open an issue on http://github.com/tinderjs/tinderjs and we'll help you out

**Comment by nicolas2bert** [comment]
@vineshhpatel @jensvdh @bfred-it  @frederocks  @sarment0  @sarment0 @vineshhpatel 
You should look at what I did really quickly 2 weeks ago.
https://github.com/nicolas2bert/tinder-stalker
I would, of course, appreciate feedbacks/contributions

Thanks

Nicolas2bert

**Comment by swagmonkey** [comment]
@mrmayfield Hey! Is this issue resolved? I'm still getting an error and would appreciate any help in this regard.

**Comment by TroyRubert** [comment]
can I get a quick ELI5 on how to get this to work?
